src/serializer/ResidualReactElementVisitor.js

```diff
@@ -154,4 +172,13 @@ export class ResidualReactElementVisitor {
     }
     return false;
   }
+
+  _getReactLibraryValue() {
+    const reactLibraryObject = this.realm.fbLibraries.react;
+    // if there is no React library, then we should throw and error
+    if (reactLibraryObject === undefined) {
+      throw new FatalError("unable to find React library reference in scope");
```

Comment:
Never just throw a FatalError without first issuing a CompilerDiagnostics. Doing so should trigger another invariant.

Reply:
If this indicates an internal error, stick to an invariant.

Reply:
Let's go back to the invariant here, it's an internal Prepack issue if it fails.

Reply:
I had assumed that this error might be thrown when the user did not `require('react')`. Guess we wouldn’t make it to this point if that were the case.